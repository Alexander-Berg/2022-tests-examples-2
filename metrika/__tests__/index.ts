import { remap } from '../normalize';

describe('remap', () => {
    it('works', () => {
        const result = remap(
            { yandexEmployeeLogin: '13' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({ yandexEmployeeLogin: 'loginFromStaff' });
    });

    it('skips keys if they not presented in raw object', () => {
        const result = remap({ yandexEmployeeLogin: '13' }, {});

        expect(result).toEqual({});
    });

    it('skips keys if they not presented in mapping object', () => {
        const result = remap({}, { 13: 'loginFromStaff' });

        expect(result).toEqual({});
    });

    it('mixed case', () => {
        const result = remap(
            { someOtherField: '42' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({});
    });
});
