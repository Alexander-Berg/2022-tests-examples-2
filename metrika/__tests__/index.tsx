import * as React from 'react';

import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Highlight, Substr } from '..';
configure({ adapter: new Adapter() });

describe('Highlight', () => {
    it('работает по строке', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">текст</Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими паттернами через пробел', () => {
        const container = render(
            <div>
                <Highlight pattern="ако екс">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с паттернами через массив', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']}>
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с подсветкой фона', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']} highlight="background">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает позиционная подсветка', () => {
        const container = render(
            <div>
                <Substr match={{ start: 1, end: 3 }}>Here comes Johny!</Substr>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
});
