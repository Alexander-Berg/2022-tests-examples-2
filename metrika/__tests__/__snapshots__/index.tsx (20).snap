// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSubstring renders filter substring 1`] = `
<styled.div>
  <Component>
    <Component>
      Укажите значение параметра
    </Component>
    <TextInput
      autocomplete={false}
      onChange={[Function]}
      size="s"
      text=""
      theme="normal"
    />
  </Component>
  <Component />
  <Component>
    <Component
      disabled={true}
      onClick={[Function]}
    />
  </Component>
</styled.div>
`;
